src/main/java/core/basesyntax/strategy/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,14 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.Transaction;
+import core.basesyntax.storage.Storage;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    @Override
+    public void apply(Transaction transaction) {
+        Fruit fruit = transaction.getFruit();
+        Integer currentQuantity = Storage.storage.get(fruit);
+        Storage.storage.put(fruit, currentQuantity - transaction.getQuantity());
```

Comment:
maybe let's throw an exception if there is no such fruit or its quantity is not enough?